### PR TITLE
Line follower for 'create_autorace' map

### PR DESCRIPTION
**Pull request by LucasJSch**
This PR targets #1.

The goal of this PR is to make the "Create" robot use the color sensors for following the lines in the 'create_autorace' map.

### CODE REVIEW

**Comment by LucasJSch** [comment]
@eborghi10 @ernestmc

**Comment by LucasJSch** [comment]
# Ready for review

**Comment by eborghi10** [comment]
@LucasJSch Please, check that you have some conflicts with your target branch.

File: ca_tools/rviz/multi_robot.rviz [hunk 1]
```diff
@@ -218,12 +220,12 @@ Visualization Manager:
           Alpha: 1
           Show Axes: false
           Show Trail: false
-        left_light_sensor_link:
+        left_color_sensor_link:
```
Comment (eborghi10) [inline ca_tools/rviz/multi_robot.rviz]: Create a new RViz config file for this purpose.

File: ca_tools/scripts/follow_lines.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                self._rotate_right()
+            self._rate.sleep()
+        rospy.spin()
+
+def main():
+    rospy.init_node('follow_lines')
+    app = FollowLines()
+    app.run()
+
+main()
```
Comment (eborghi10) [inline ca_tools/scripts/follow_lines.py]: Again, put an empty line at the end.

File: ca_tools/scripts/follow_lines.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+# -*- coding: utf-8 -*-
+
+from threading import Lock
+
+import rospy
+from enum import Enum
+from geometry_msgs.msg import Twist, Pose
+from std_msgs.msg import Bool, Int64
+
+class RobotState(Enum):
```
Comment (eborghi10) [inline ca_tools/scripts/follow_lines.py]: It will be easier if you use [smach_ros](http://wiki.ros.org/smach_ros).

**Comment by LucasJSch** [comment]
# Ready for review.

Note: I didn't implement it with `smach_ros`. I don't like the approach of the library for implementing state machines. If you (@ernestmc & @eborghi10) consider this to be mandatory for us to use in ROS code (because of conventions maybe?), I'll implement it.

**Comment by eborghi10** [comment]
This PR has conflicts :fearful:

**Comment by LucasJSch** [comment]
> This PR has conflicts **:fearful:**

Solved conflicts **:+1:**

**Pull request merged**